content/articles/how-to-ship-zero-javascript-with-astro/index.md

```diff
@@ -0,0 +1,464 @@
+ 
+* class component
+ 
+* functional component
+ 
+For this article, we will be using a functional component.
+ 
+Navigate to the component folder in the main project folder. Create a **Tour.jsx** file in the component folder.
+ 
+Inside the **Tour.jsx** file you created add the following:
+ 
+```
+ 
+import React from 'React';
+ 
+const Tours = ({ tours }) => {
+ 
+const [readMore, setReadMore] = React.useState(false)
+ 
+return(
+ 
+<step>
+ 
+<div className="title">
+ 
+  <h2>Our Tours</h2>
+ 
+</div>
+ 
+<div>
+ 
+  {
+ 
+    tours.map((tour) => {
+ 
+      return (
+ 
+          <article className="single-tour">
+ 
+            <img width='400' src={tour.image} alt={tour.name} />
+ 
+            <footer>
+ 
+              <div className="tour-info">
+ 
+                <h4>{tour.name}</h4>
+ 
+                <h4 className="tour-price">${tour.price}</h4>
+ 
+              </div>
+ 
+              <p>
+ 
+                {tour.info}
+ 
+              </p>
+ 
+            </footer>
+ 
+          </article>
+ 
+      )
+ 
+    })
+ 
+  }
+ 
+</div>
+ 
+</step>
+ 
+)
+ 
+};
+ 
+export default Tours;
+ 
+```
+ 
+The above code is a function component named **Tours**, we passed in a prop called **tours** as a parameter. The above **Tours** function will display your data in the web page.
+ 
+Inside the **Tours** functional component, we have a bunch of **jsx** code that we will explain below:
+ 
+* **tours.map**: This maps the **tours**  passed in as a prop.
+ 
+* **const [readMore, setReadMore] = React.useState(false)**: This specifies two state variables. **readMore** has the current value of false.  **setReadMore** enables you to change the current value. You won't be needing this now until later step.
+ 
+* **return**: This specifies what should be rendered to the browser. It contains a bunch of **HTML** structures.
+ 
+* **tour.image**: This specifies the variable named image in each element of the  **tours** prop.
+ 
+* **tour.name**: This specifies the variable called name in each element of the **tours** prop.
+ 
+* **tour.price**: This specifies the variable named price in each element of the**tours** prop.
+ 
+* **tour.info**: This specifies the variable named info in each element of the **tours** prop.
+ 
+* **export default Tours** : This exports the **Tours.jsx** component. This makes it to be reusable in other components.
+ 
+### step 6 - Using Astro to Fetch and Pass Data to Your React Component
+ 
+In this step, you will learn how to fetch and pass data to
+ 
+the **Tours.jsx** component you created above.
+ 
+Navigate into **Tour.astro** file, at the top you will see three dashes above and below. Inside the dashes is where we will write some javascript code that will fetch our data as seen below:
+ 
+Copy the code below inside **Tour.astro** file:
+ 
+```
+ 
+---
+ 
+import { Markdown } from 'astro/components';
+ 
+import  Tours  from '../components/Tours.jsx';
+ 
+const url = 'https://course-api.com/React-tours-project'
```

Comment:
Could you use backticks on the entire string as `const url = 'https://course-api.com/React-tours-project'` to avoid it being rendered as a web link?

Reply:
> Could you use backticks (``) on the entire string as `const url = 'https://course-api.com/React-tours-project'` to avoid it being rendered as a web link?

@Qodestackr done boss. thanks man.